mail_debrand/models/mail_thread.py

```diff
@@ -10,5 +10,13 @@ class MailThread(models.AbstractModel):
 
     def _replace_local_links(self, html, base_url=None):
         html = super()._replace_local_links(html, base_url=base_url)
-        html_debranded = self.env["mail.template"]._debrand_body(html)
+        lang = False
+        if {'default_template_id', 'default_model', 'default_res_id'} \
+                <= self.env.context.keys():
+            template = self.env["mail.template"].browse(
+                self.env.context['default_template_id'])
+            lang = template._render_template(template.lang,
+                                             self.env.context['default_model'],
+                                             self.env.context['default_res_id'])
```

Comment:
Trying this I noticed that the correct `lang` is already in the context because `generate_email` puts it there: https://github.com/odoo/odoo/blob/2be47637dd5cc48ca561ff85cf2d359dde0829ab/addons/mail/models/mail_template.py#L443
so looks like there is no need to look for these values being in `context`, find the template and call render.
This would be much cleaner if we could just use the language that is already in the context, what do you think?

If any test has a problem with that, that's probably because the test does not emulate correctly what really happens.